### ISSUE TITLE
Katex (Latex) not compatible with Gitlab

### ISSUE THREAD

**Issue by toger5** [yzhang-gh/vscode-markdown#570]
Git lab requires $`19+1=20`$ but markdown all in one does not.

**Comment by yzhang-gh** [comment]
There is no unified syntax for Math (in Markdown) so we probably can't have a good solution on this.

**Comment by toger5** [comment]
Is it possible to provide it as an option in the vs code markdown renderer? `math-syntax-flavor = {gitlab | basic}`

**Comment by yzhang-gh** [comment]
It is a bit hard. We are currently relying on [`markdown-it-katex`](https://github.com/waylonflinn/markdown-it-katex) for math rendering.

**Comment by NeroBurner** [comment]
maybe we could switch to [markdown-it-texmath](https://github.com/goessner/markdown-it-texmath)

the `texmath` project provides a possibility to select the delimiter one wants to use

![Screenshot_20200309_163919](https://user-images.githubusercontent.com/9076163/76230812-8cdf2380-6224-11ea-805b-40ba187298c8.png)

on a fast check the `texmath` repo has more recent updates:
- latest update on `markdown-it-texmath` was 5 month ago ([latest commit as of writing](https://github.com/goessner/markdown-it-texmath/commit
- latest update on `markdown-it-katex` was 4 years ago ([latest commit as of writing](https://github.com/waylonflinn/markdown-it-katex/commit))/81b84a7d91d74a1ba3d4769b847a5bf1947256cb
/78c548829ce2ef85c73dc71e680d01e5ae41ffbf/78c548829ce2ef85c73dc71e680d01e5ae41ffbf))

**Comment by yzhang-gh** [comment]
I would suggest [Markdown+Math extension](https://marketplace.visualstudio.com/items?itemName=goessner.mdmath), from the author of `markdown-it-texmath` 😂.

**Comment by NeroBurner** [comment]
maybe invite @goessner to work together on one extension using `markdown-it-texmath`. Then you'd have one more developer, one less competing extension, more user option (specify inline math delimiter) and a very good connection to the backend developer

just a suggestion :man_shrugging:

**Comment by yzhang-gh** [comment]
I'm open to the collaboration of course. It just looks like Markdown+Math has already covered its target users (who need better math support), while this extension targets common Markdown users.

**Comment by toger5** [comment]
what about users who want math support and the features if vscode-markdown. are they compatible?

**Comment by yzhang-gh** [comment]
I have added an option `math.enabled`. They should be compatible now (in terms of the VSCode's builtin preview).

(the [dev build](https://github.com/yzhang-gh/vscode-markdown#latest-development-build))

**Comment by yxs** [comment]
dev build is not accessible. 
`Project not found or access denied.` 
I try to access other projects such as https://ci.appveyor.com/project/benvanik/xenia and it's accessible.

**Comment by yzhang-gh** [comment]
It has been moved to https://github.com/yzhang-gh/vscode-markdown/actions/runs/64661491.

**Comment by goessner** [comment]
I implemented [markdown-it-texmath](https://github.com/goessner/markdown-it-texmath) intentionally as a standalone plugin for `markdown-it-texmath` usable with `node.js` and in browsers.

So if it serves the needs of you and your users ... go ahead and take it.

Thanks for your interest.

**Comment by yzhang-gh** [comment]
Thanks you @goessner. Your extension provides a very nice support of math in Markdown. Let's first see whether our two extensions can work seamlessly now. Further, we can also work together if integration is needed for better user experience.

**Comment by goessner** [comment]
Hmm ... I am totally happy with my `mdmath` vscode extension, so it doesn't need to work together with your extension from my point of view.

I only make an offer, that you can use my `markdown-it` plugin `markdown-it-texmath` in your vscode extension ... that's all. Of course you need to test for possible conflicts.

thanks

**Comment by yzhang-gh** [comment]
> Hmm ... I am totally happy with my `mdmath` vscode extension, so it doesn't need to work together with your extension from my point of view.

I totally agree. I just wanted to say I'm willing to do something _**if**_ collaboration would be needed 😅.
I've made some changes to my plugin so that some users can use this one in parallel to `mdmath`. That's the solution I suggested earlier under this issue.
Sorry for the confusion.

**Comment by digitalmoksha** [comment]
Just ran across this issue. Just fyi, we're working on adding `$` and `$$` support in GitLab, https://gitlab.com/gitlab-org/gitlab/-/issues/21757
